components/script_bindings/codegen/CodegenRust.py

```diff
@@ -2580,26 +2586,45 @@ def __init__(self, name):
         CGGeneric.__init__(self, f"{name.replace('<D>', '::<D>')}::new(cx, ${{val}}.get().to_object())")
 
 
-def getAllTypes(descriptors, dictionaries, callbacks, typedefs):
+def getAllTypes(
+    descriptors: List[Descriptor],
+    dictionaries: List[IDLDictionary],
+    callbacks: List[IDLCallback],
+    typedefs: List[IDLTypedef]
+) -> Generator[Tuple[IDLType, Optional[Descriptor]], None, None]:
     """
     Generate all the types we're dealing with.  For each type, a tuple
     containing type, descriptor, dictionary is yielded.  The
     descriptor can be None if the type does not come from a descriptor.
     """
     for d in descriptors:
         for t in getTypesFromDescriptor(d):
+            if t.isRecord():
+                yield (t.inner, d)
             yield (t, d)
     for dictionary in dictionaries:
         for t in getTypesFromDictionary(dictionary):
+            if t.isRecord():
+                yield (t.inner, None)
             yield (t, None)
     for callback in callbacks:
         for t in getTypesFromCallback(callback):
+            if t.isRecord():
+                yield (t.inner, None)
             yield (t, None)
     for typedef in typedefs:
+        if typedef.innerType.isRecord():
+            yield (typedef.innerType.inner, None)
```

Comment:
I am wondering if we should just patch webidl.py: https://github.com/servo/servo/issues/37038#issuecomment-2888467476

Reply:
I tried doing that but it feels wrong. As I understand it, `unroll` seems to only be used for types that "are" another type - like typedefs and wrapper types. In these cases, we never care the not-unrolled type, because we can always treat it as its unrolled version. `record` on the other hand exists on its own as an independent type.

(Patching webidl as you described does work though)